benchopt/base.py

```diff
@@ -94,11 +94,20 @@ def run(self, stop_val):
         This function should not return the parameters which will be
         retrieved by a subsequent call to get_result.
 
+        If `stop_strategy` is set to `"callback"`, then `run` should call the
+        callback at each iteration. The callback will compute the time,
+        the objective function and store relevant quantities for BenchOpt.
+        Else, the `stop_val` parameter should be specified.
+
         Parameters
         ----------
-        stop_val : int | float
+        stop_val : int | float | callable
             Value for the stopping criterion of the solver for. It allows to
             sample the time/accuracy curve in the benchmark.
+            If it is a callable, then it should act as a callback. This
+            callback should be called once for each iteration with argument
+            the current iterate `x`. The callback returns False when the
```

Comment:
We called this `parameters` in the `get_result`. I would stick to this or change it there too.
```suggestion
            the current iterate `parameters`. The callback returns False when the
```